src/Build.UnitTests/ChangeWaves_Tests.cs

```diff
@@ -0,0 +1,319 @@
+                    p.Build().ShouldBeTrue();
+
+                    BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                    log.AssertLogContains("out of rotation");
+                    log.AssertLogContains("Hello World!");
+                }
+            }
+        }
+
+        [Fact]
+        public void VersionSetToValidValueButInvalidVersionSetsNextVersion()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                env.SetChangeWave($"{ChangeWaves.LowestWaveAsVersion.Major}.{ChangeWaves.LowestWaveAsVersion.Minor}.{ChangeWaves.LowestWaveAsVersion.Build+1}");
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                // All waves should be disabled
+                    string projectFile = $"" +
+                        $"<Project>" +
+                            $"<Target Name='HelloWorld' Condition=\"'$(MSBUILDDISABLEFEATURESFROMVERSION)' == '{ChangeWaves.AllWaves[1]}' and $([MSBuild]::AreFeaturesEnabled('{ChangeWaves.LowestWave}'))\">" +
+                                $"<Message Text='Hello World!'/>" +
+                            $"</Target>" +
+                        $"</Project>";
+
+                    TransientTestFile file = env.CreateFile("proj.csproj", projectFile);
+
+                    ProjectCollection collection = new ProjectCollection();
+                    MockLogger log = new MockLogger();
+                    collection.RegisterLogger(log);
+
+                    Project p = collection.LoadProject(file.Path);
+                    p.Build().ShouldBeTrue();
+
+                    BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+                    log.AssertLogContains("Hello World!");
+
+            }
+        }
+
+        [Fact]
+        public void CorrectlyDetermineEnabledFeatures()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                env.SetChangeWave(ChangeWaves.HighestWave);
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                for (int i = 0; i < ChangeWaves.AllWaves.Length-1; i++)
+                {
+                    ChangeWaves.DisabledWave = null;
+                    ChangeWaves.AreFeaturesEnabled(ChangeWaves.AllWaves[i]).ShouldBe(true);
+
+                    string projectFile = $"" +
+                        $"<Project>" +
+                            $"<Target Name='HelloWorld' Condition=\"$([MSBuild]::AreFeaturesEnabled('{ChangeWaves.AllWaves[i]}'))\">" +
+                                $"<Message Text='Hello World!'/>" +
+                            $"</Target>" +
+                        $"</Project>";
+
+                    TransientTestFile file = env.CreateFile("proj.csproj", projectFile);
+
+                    ProjectCollection collection = new ProjectCollection();
+                    MockLogger log = new MockLogger();
```

Comment:
Also check whether it's correctly set for calling your API from C# (for this and the next test)

Reply:
I'm not sure what there is to check? Something like `ChangeWaves.DisabledVersion.ShouldBe(blah)`?